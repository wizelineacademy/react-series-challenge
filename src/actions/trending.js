const TRENDING_GET = 'TRENDING_GET'
const TRENDING_FETCHED = 'TRENDING_FETCHED'
const TRENDING_FAILED = 'TRENDING_FAILED'

const getTrending = payload => ({ type: TRENDING_GET, payload })
const fetchedTrending = payload => ({ type: TRENDING_FETCHED, payload })
const failedTrending = payload => ({ type: TRENDING_FAILED, payload })

export default {
    types: {
        TRENDING_GET,
        TRENDING_FETCHED,
        TRENDING_FAILED
    },
    creators: {
        getTrending,
        fetchedTrending,
        failedTrending
    }
}