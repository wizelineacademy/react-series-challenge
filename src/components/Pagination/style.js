import styled from 'styled-components'

export const PaginationButton = styled.span`
  color: black;
  background-color: white;
  padding: 8px 16px;
  text-decoration: none;
  transition: background-color .3s;
  border: 1px solid #ddd;
  cursor: pointer;
  margin-top: 30px;
  ${ props => props.active && 
    `background-color: #4CAF50;
    color: white;
    border: 1px solid #4CAF50;
    cursor: default;
    `
  }
`