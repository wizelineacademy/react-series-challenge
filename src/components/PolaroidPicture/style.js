import styled from 'styled-components'
import { Link } from 'react-router-dom';


export const PolaroidListItem = styled.li`
  position: relative;
  width: ${props => props.isSingleView ? '500px;' : '200px;'}
  height: ${props => props.isSingleView ? '650px;' : '250px;'}
  display: inline-block;
  padding: 10px;
  margin: 10px;
  background-color: white;
  box-shadow: 7px 7px 4px black;
  border: 1px solid #000
  img {
    max-width: 100%;
    height: 80%;
  }
`
export const PolaroidLink = styled(Link)`
  color: white;
`

export const PolaroidStar = styled.div`
  background-image: url('/favorited.ico');
  background-size: contain;
  width: 50px;
  height: 50px;
  position: absolute;
  top: -25px;
  right: calc(50% - 25px);
`

export const PolaroidButton = styled.button`
  margin: 5px;
  padding: 10px 19px;
  margin: 0px 10px -10px 0px;
  border-radius: 10px;
  font-family: 'Pacifico', cursive;
  font-size: 17px;
  color: #FFF;
  text-decoration: none; 
  border: none;
  outline:none;
  cursor: pointer;
  display: none;
  ${ props => props.color === 'blue' ?
      `
        background-color: #3498DB;
        text-shadow: 0px -2px #2980B9;
      `
      :
      `
        background-color: #82BF56;
        text-shadow: 0px -2px #669644;
        float: right;
      `
  }
  :active {
    transform: translate(0px,5px);
    -webkit-transform: translate(0px,5px);
    border-bottom: 1px solid;
  }
`

export const PolaroidLoading = styled.span`
  position: absolute;
  width: 200px;
  height: 200px;
  left: 10px;
  background-image: url("/loader.svg");
  background-size: contain;
  background-color: black;
`