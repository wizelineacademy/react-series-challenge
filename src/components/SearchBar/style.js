import styled from 'styled-components'

export const Container = styled.div `
  display: flex;
  justify-content: center;
  margin-bottom: 70px;
`

export const SearchBox = styled.div `
  position: absolute;
  input {
    border: none;
    background: none;
    z-index: 1;
    width: 50px;
    height: 50px;
    transition: all .25s ease-in .25s;
    color: transparent;
    font-size: 1.25rem;
    line-height:50px;
    &:hover {
      cursor: pointer;
      &:focus {
        cursor: text;
      }
      + span {
        background: rgba(255,255,255,0.2);
      }
    }
    &:focus {
      width: 200px;
      padding: 0 10px;
      outline: none;
      color: black;
      background:none;
      color: white;
      + span {
        width: 200px;
        &::before {
          width:2px;
          opacity: 0;
          transition: all .25s ease-in;
        }
      }
    }
    + span {
      z-index: -1;
      position: absolute;
      border: 2px solid white;
      top: 0;
      width: 50px;
      height: 50px;
      transition: all .25s ease-in .25s;
      border-radius: 50px;
      left: 0;
      &::before {
        transition: all .25s ease-in .5s;
        transform-origin: left top;
        content: '';
        position: absolute;
        width: 20px;
        height: 7px;
        border-radius: 5px;
        background: white;
        transform: rotate(45deg) translate(43px, 10px);
      }
    }
  }
`