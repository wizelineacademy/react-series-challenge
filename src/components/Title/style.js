import styled from 'styled-components'

export const BurningEffect = styled.h1`
  color: #fff;
  text-shadow: 0px -1px 4px white, 0px -2px 10px yellow, 0px -10px 20px #ff8000, 0px -18px 40px red;
  font: 2em 'BlackJackRegular';
`

export const NeonEffect = styled.h1`
  color: #fff;
  text-shadow: 0 0 5px #fff, 0 0 10px #fff, 0 0 15px #fff, 0 0 20px #ff2d95, 0 0 30px #ff2d95, 0 0 40px #ff2d95, 0 0 50px #ff2d95, 0 0 75px #ff2d95;
  letter-spacing: 5px;
  font: 3em 'Miso';
`