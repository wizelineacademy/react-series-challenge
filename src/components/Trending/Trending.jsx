import React from 'react';
import PropTypes from 'prop-types';

import Template from '../Template';
import NavBar from '../NavBar';
import GifGrid from '../GifGrid';

class Trending extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  componentDidMount() {
    const { fetchTrending, fetchFavorite } = this.props;
    fetchTrending();
    fetchFavorite();
  }

  handleChange(event) {
    const { searchTrending, fetchTrending } = this.props;
    const { value } = event.target;
    if (value !== '') {
      searchTrending({ payload: { text: value } });
    } else {
      fetchTrending();
    }
  }

  render() {
    const {
      trendingGifs,
      favoriteGifs,
      filterTrendingText,
    } = this.props;

    const links = [
      { text: 'Trending Gifs', url: '/trending', isActive: true },
      { text: 'Favorite Gifs', url: '/favorite', isActive: false },
    ];
    return (
      <Template>
        <NavBar links={links} placeholder="Search..." type="text" value={filterTrendingText} onChange={this.handleChange} />
        <div>
          <GifGrid
            gifs={trendingGifs}
            favoriteGifs={favoriteGifs}
            filterText={filterTrendingText}
          />
        </div>
      </Template>
    );
  }
}

Trending.propTypes = {
  filterTrendingText: PropTypes.string.isRequired,
  fetchTrending: PropTypes.func.isRequired,
  searchTrending: PropTypes.func.isRequired,
  fetchFavorite: PropTypes.func.isRequired,
  trendingGifs: PropTypes.object.isRequired,
  favoriteGifs: PropTypes.object.isRequired,
};

export default Trending;
