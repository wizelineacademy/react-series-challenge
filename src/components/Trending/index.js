import trending from '../../actions/trending'
import { bindActionCreators } from 'redux';
import { Title, Gifs, Gif } from './trending.style'
import React, { Component, Fragment } from 'react'
import { connect } from 'react-redux'

class Trending extends Component {

    componentWillMount() {
        const { getTrending } = this.props
        getTrending()
    }


    render() {
        /**
         * 
         */
        const isFavorite = gif => this.props.favorites.some(favorite => favorite.id === gif.id)
        const gifs = this.props.trending.map(gif => <Gif favorite={isFavorite(gif)} key={gif.id} gif={gif} src={gif.images.downsized.url} alt={gif.title} />)
        /**
         * 
         */
        return (
            <Fragment>
                <Title> Trending </Title>
                <Gifs>
                    { gifs }
                </Gifs>
            </Fragment>
        )
    }
}

const mapStateToProps = ({ trending, favorites: { items: favorites } }) => {
    return { trending, favorites }
}

const mapDispatchToProps = dispatch => {
    /**
     *
     */
    const { creators: { getTrending } } = trending
    /**
     *
     */
    return bindActionCreators({ getTrending }, dispatch)
}

export { Trending }
export default connect(mapStateToProps, mapDispatchToProps)(Trending)

