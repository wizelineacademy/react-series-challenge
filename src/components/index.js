import Card from './Card'
import Detail from './Detail'
import GifItem from './GifItem'
import Grid from './Grid'
import Header from './Header'
import Home from './Home'
import SearchBar from './SearchBar'

export {
  Card,
  Detail,
  GifItem,
  Grid,
  Header,
  Home,
  SearchBar
}
