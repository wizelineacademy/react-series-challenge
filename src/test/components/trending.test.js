import { Trending } from '../../components/Trending'
import { Title, Gifs, Gif } from '../../components/Trending/trending.style'
import { shallow, simulate } from 'enzyme'
import 'jest-styled-components'
import React from 'react';
import Store from '../../store'

const items = [{
    src: 'https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif',
    alt: 'BMO',
    onClick: jest.fn(),
    favorite: false,
    images: {
        downsized: {
            url: 'https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif'
        }
    }

}, {
    src: 'https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif',
    alt: 'BMO',
    onClick: jest.fn(),
    favorite: false,
    images: {
        downsized: {
            url: 'https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif'
        }
    }
}]
const props = {
    getTrending: jest.fn(),
    favorites: items,
    trending: items
}

describe('Trending', () => {
    
    test('Title styled', () => {
        const component = shallow(<Title > Hola </Title>)
        expect(component).toHaveStyleRule('font-size', '2rem')
        expect(component).toHaveStyleRule('font-family', 'sans-serif')
        expect(component).toHaveStyleRule('text-align', 'center')
    })

    test('Gifs styled', () => {
        const component = shallow(<Gifs />)
        expect(component).toHaveStyleRule('display', 'grid')
        expect(component).toHaveStyleRule('grid-gap', '0.5rem')
        expect(component).toHaveStyleRule('justify-content', 'center')
        expect(component).toHaveStyleRule('grid-template-columns', 'repeat(auto-fill,12rem)')
        expect(component).toHaveStyleRule('grid-template-rows', 'repeat(auto-fill,12rem)')
    })

    test('Trending', () => {
        const component = shallow(<Trending {...props} />)
        expect(component).toMatchSnapshot();
    })

    test('Trending Gif', () => {
        const component = shallow(<Trending {...props} />)
        expect(component.find(Gif)).toHaveLength(2);
    })

    test('Trending Gis', () => {
        const component = shallow(<Trending {...props} />)
        expect(component.find(Gifs)).toHaveLength(1);
    })

    test('Trending Title', () => {
        const component = shallow(<Trending {...props} />)
        expect(component.find(Title)).toHaveLength(1);
    })

    test('Trending Title', () => {
        const component = shallow(<Trending {...props} />)
        expect(props.getTrending).toHaveBeenCalled();
    })

})