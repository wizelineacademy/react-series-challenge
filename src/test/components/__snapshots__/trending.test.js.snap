// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trending Trending 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Trending
    favorites={
        Array [
            Object {
              "alt": "BMO",
              "favorite": false,
              "images": Object {
                "downsized": Object {
                  "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                },
              },
              "onClick": [Function],
              "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
            },
            Object {
              "alt": "BMO",
              "favorite": false,
              "images": Object {
                "downsized": Object {
                  "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                },
              },
              "onClick": [Function],
              "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
            },
          ]
    }
    getTrending={[Function]}
    trending={
        Array [
            Object {
              "alt": "BMO",
              "favorite": false,
              "images": Object {
                "downsized": Object {
                  "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                },
              },
              "onClick": [Function],
              "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
            },
            Object {
              "alt": "BMO",
              "favorite": false,
              "images": Object {
                "downsized": Object {
                  "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                },
              },
              "onClick": [Function],
              "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <styled.h1>
           Trending 
</styled.h1>,
        <styled.div>
          <Styled(Connect(Component))
                    alt={undefined}
                    favorite={true}
                    gif={
                              Object {
                                        "alt": "BMO",
                                        "favorite": false,
                                        "images": Object {
                                          "downsized": Object {
                                            "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                          },
                                        },
                                        "onClick": [Function],
                                        "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                      }
                    }
                    src="https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif"
          />
          <Styled(Connect(Component))
                    alt={undefined}
                    favorite={true}
                    gif={
                              Object {
                                        "alt": "BMO",
                                        "favorite": false,
                                        "images": Object {
                                          "downsized": Object {
                                            "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                          },
                                        },
                                        "onClick": [Function],
                                        "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                      }
                    }
                    src="https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif"
          />
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": " Trending ",
        },
        "ref": null,
        "rendered": " Trending ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Connect(Component))
              alt={undefined}
              favorite={true}
              gif={
                            Object {
                                          "alt": "BMO",
                                          "favorite": false,
                                          "images": Object {
                                            "downsized": Object {
                                              "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                            },
                                          },
                                          "onClick": [Function],
                                          "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                        }
              }
              src="https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif"
/>,
            <Styled(Connect(Component))
              alt={undefined}
              favorite={true}
              gif={
                            Object {
                                          "alt": "BMO",
                                          "favorite": false,
                                          "images": Object {
                                            "downsized": Object {
                                              "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                            },
                                          },
                                          "onClick": [Function],
                                          "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                        }
              }
              src="https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": undefined,
              "favorite": true,
              "gif": Object {
                "alt": "BMO",
                "favorite": false,
                "images": Object {
                  "downsized": Object {
                    "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                  },
                },
                "onClick": [Function],
                "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
              },
              "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": undefined,
              "favorite": true,
              "gif": Object {
                "alt": "BMO",
                "favorite": false,
                "images": Object {
                  "downsized": Object {
                    "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                  },
                },
                "onClick": [Function],
                "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
              },
              "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <styled.h1>
             Trending 
</styled.h1>,
          <styled.div>
            <Styled(Connect(Component))
                        alt={undefined}
                        favorite={true}
                        gif={
                                    Object {
                                                "alt": "BMO",
                                                "favorite": false,
                                                "images": Object {
                                                  "downsized": Object {
                                                    "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                                  },
                                                },
                                                "onClick": [Function],
                                                "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                              }
                        }
                        src="https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif"
            />
            <Styled(Connect(Component))
                        alt={undefined}
                        favorite={true}
                        gif={
                                    Object {
                                                "alt": "BMO",
                                                "favorite": false,
                                                "images": Object {
                                                  "downsized": Object {
                                                    "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                                  },
                                                },
                                                "onClick": [Function],
                                                "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                              }
                        }
                        src="https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif"
            />
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": " Trending ",
          },
          "ref": null,
          "rendered": " Trending ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Connect(Component))
                alt={undefined}
                favorite={true}
                gif={
                                Object {
                                                "alt": "BMO",
                                                "favorite": false,
                                                "images": Object {
                                                  "downsized": Object {
                                                    "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                                  },
                                                },
                                                "onClick": [Function],
                                                "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                              }
                }
                src="https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif"
/>,
              <Styled(Connect(Component))
                alt={undefined}
                favorite={true}
                gif={
                                Object {
                                                "alt": "BMO",
                                                "favorite": false,
                                                "images": Object {
                                                  "downsized": Object {
                                                    "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                                  },
                                                },
                                                "onClick": [Function],
                                                "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                                              }
                }
                src="https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": undefined,
                "favorite": true,
                "gif": Object {
                  "alt": "BMO",
                  "favorite": false,
                  "images": Object {
                    "downsized": Object {
                      "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                    },
                  },
                  "onClick": [Function],
                  "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                },
                "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": undefined,
                "favorite": true,
                "gif": Object {
                  "alt": "BMO",
                  "favorite": false,
                  "images": Object {
                    "downsized": Object {
                      "url": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                    },
                  },
                  "onClick": [Function],
                  "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
                },
                "src": "https://media3.giphy.com/media/UdtJqVG2gCchy/giphy-downsized.gif",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
